.gitlab-ci.yml

```diff
@@ -3,225 +3,78 @@ stages:
-        -DGINKGO_BUILD_TESTS=ON -DGINKGO_BUILD_EXAMPLES=ON
-        -DGINKGO_FAST_TESTS=${FAST_TESTS}
-        -DGINKGO_MIXED_PRECISION=${MIXED_PRECISION}
-        -DGINKGO_CONFIG_LOG_DETAILED=${CONFIG_LOG}
-        -DGINKGO_DPCPP_SINGLE_MODE=${DPCPP_SINGLE_MODE}
-        -DGINKGO_RUN_EXAMPLES=${RUN_EXAMPLES}
-        -DGINKGO_EXPORT_BUILD_DIR=${EXPORT_BUILD_DIR}
-    - ninja -j${NUM_CORES} -l${CI_LOAD_LIMIT} install
-    - |
-        (( $(ctest -N | tail -1 | sed 's/Total Tests: //') != 0 )) || exit 1
-    - ctest -V
-    - ninja test_install
-    - pushd test/test_install
-    - ninja install
-    - popd
-    - |
-      if [ "${RUN_EXAMPLES}" == "ON" ]; then
-         export EX_ARG="reference"
-         ninja run_all_examples
-         ninja validate_all_examples
-         if [ "{BUILD_OMP}" == "ON" ]; then
-            export EX_ARG="omp"
-            ninja run_all_examples
-            ninja validate_all_examples
-        fi
-        if [ "{BUILD_CUDA}" == "ON" ]; then
-            export EX_ARG="cuda"
-            ninja run_all_examples
-            ninja validate_all_examples
-        fi
-        if [ "{BUILD_HIP}" == "ON" ]; then
-            export EX_ARG="hip"
-            ninja run_all_examples
-            ninja validate_all_examples
-        fi
-      fi
-    - if [ -n "${SYCL_DEVICE_TYPE}" ]; then unset SYCL_DEVICE_TYPE; fi
-    - if [ -n "${SYCL_DEVICE_FILTER}" ]; then unset SYCL_DEVICE_FILTER; fi
-    - if [ "${EXPORT_BUILD_DIR}" == "ON" ]; then ninja test_exportbuild; fi
-  dependencies: []
-  except:
-      - schedules
 
 status_pending:
   stage: init-status
   extends:
+    - .default_variables
     - .pr_condition
-    - .use_gko-nocuda-openmpi-gnu9-llvm8
+    - .use_status-job-settings
   variables:
     STATUS_CONTEXT: "quick"
   script: |
-    curl -X POST -H "Accept: application/vnd.github.v3+json" -H "Authorization: token ${BOT_STATUS_TOKEN}" \
+    curl -s -X POST -H "Accept: application/vnd.github.v3+json" -H "Authorization: token ${BOT_STATUS_TOKEN}" \
     https://api.github.com/repos/ginkgo-project/ginkgo/statuses/${CI_COMMIT_SHA} \
     -d "{\"state\":\"pending\",\"context\":\"ci/gitlab/${STATUS_CONTEXT}\",\"target_url\":\"${CI_PIPELINE_URL}\"}"
 
 status_success:
   stage: finalize-status
   extends:
+    - .default_variables
     - .pr_condition
-    - .use_gko-nocuda-openmpi-gnu9-llvm8
+    - .use_status-job-settings
   variables:
     STATUS_CONTEXT: "quick"
   # we always exit with the code 3 such that it will process when retrying
   script: |
-    curl -X POST -H "Accept: application/vnd.github.v3+json" -H "Authorization: token ${BOT_STATUS_TOKEN}" \
+    curl -s -X POST -H "Accept: application/vnd.github.v3+json" -H "Authorization: token ${BOT_STATUS_TOKEN}" \
     https://api.github.com/repos/ginkgo-project/ginkgo/statuses/${CI_COMMIT_SHA} \
     -d "{\"state\":\"success\",\"context\":\"ci/gitlab/${STATUS_CONTEXT}\",\"target_url\":\"${CI_PIPELINE_URL}\"}"
-    exit 3
```

Comment:
This was put in here deliberately by Mike, since it allows us to rerun the entire pipeline after it finished, by just clicking "rerun".